sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1585 @@
+and any necessary `device`.
+
+Available only if `propertyListA` and `propertyListB` are specializations of the `properties` class.
+
+
+|====
+
+=== USM Memory Allocation Properties
+
+This section specifies the properties that can be passed to the USM memory allocation functions with `properties` support defined in Section <<section.usm.allocs>>.
+A SYCL implementation or SYCL backend may support additional properties other than those defined here, provided they are defined in accordance with the
+link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties] extension.
+If unsupported properties are passed to the USM memory allocation functions with `properties` support, the compiler will issue a diagnostic error.
+
+All properties supported by `annotated_ptr` are supported by the USM memory allocation functions with `properties` support. These properties do not affect the behavior of the allocation functions at runtime, except where explicitly specified.
+
+
+[NOTE]
+====
+The USM memory allocation functions with `properties` support defined in this extension support all properties supported by `annotated_ptr`, as a convenience. This allows the user to attach all desired properties to the `annotated_ptr` in a single step.
+====
+
+Table <<table.usm.malloc.properties>> lists properties that affect the behavior of the allocation functions.
+
+[[table.usm.malloc.properties]]
+.Properties that affect the behavior of the USM memory allocation functions with properties support
+[options="header"]
+|====
+|Property|Description|Source
+a|
+sycl::ext::oneapi::experimental::alignment
+| 	
+If this property is passed to a USM memory allocation function with `properties` support, it instructs the runtime to allocate memory with this alignment in bytes.
+|
+link:../proposed/sycl_ext_oneapi_annotated_ptr.asciidoc[sycl_ext_oneapi_annotated_ptr]
+|====
+
+Table <<table.usm.malloc.devhostsh>> lists the new properties introduced by this extension.
+
+[[table.usm.malloc.devhostsh]]
+.New compile-time constant properties introduced by this extension
+[options="header"]
+|====
+|Property|Values|Description|Supported By
+a|
+sycl::ext::oneapi::experimental::usm_kind
+| 	
+sycl::usm::alloc::host
+sycl::usm::alloc::device
+sycl::usm::alloc::shared
+|
+Indicates the kind of USM memory accessed by dereferencing this pointer and pointers derived from this pointer.
+Instructs the runtime to allocate USM memory of this kind.
+|
+`annotated_ptr`, +
+and the USM memory allocation functions with `properties` support defined in this extension.
+|====
+
+
+The `sycl::ext::oneapi::experimental::usm_kind` property is a compile-time constant property with a single non-type parameter. This parameter is a value belonging to the enumeration `sycl::usm::alloc`.
+The `sycl::ext::oneapi::experimental::usm_kind` property is supported by `annotated_ptr` and the USM memory allocation functions defined in this extension.
+
+=== Deallocation
```

Comment:
Should we add an overload of sycl::free that takes an annotated_ptr to make this easier?

Reply:
Done.